src/qhttp/Response.cc

```diff
@@ -182,13 +164,41 @@ std::string Response::_headers() const
 
     auto r = responseStringsByCode.find(status);
     if (r == responseStringsByCode.end()) r = responseStringsByCode.find(500);
-    headerst << r->first << " " << r->second << "\r\n";
+    headerst << r->first << " " << r->second;
 
+    auto ilength = headers.find("Content-Length");
+    std::size_t length = (ilength == headers.end()) ? 0 : std::stoi(ilength->second);
+    LOGLS_INFO(_log, logger(_server) << logger(_socket) << headerst.str() << " + " << length << " bytes");
+
+    headerst << "\r\n";
     for(auto const& h : headers) {
         headerst << h.first << ": " << h.second << "\r\n";
     }
 
     return headerst.str();
 }
 
+
+void Response::_write()
+{
+    if (_transmissionStarted.test_and_set()) {
+        LOGLS_WARN(_log, logger(_server) << logger(_socket)
+            << "handler logic error: multiple responses sent");
```

Comment:
This would occur only if a user of the `qhttp` library programmed their custom handler code incorrectly (e.g. calling a response send method multiple times, or throwing an exception after already calling a response `send` method.)  It should not ever occur in practice, but a `qhttp` user could always make mistakes when writing their handlers.

The atomic is here specifically to provide a safeguard against bad user code, and prevents `qhttp` from sending an invalid "double response."  The (new) warning means we can now see it in the logs if it ever _does_ happen.

Reply:
Perhaps this should be `LOGLS_ERROR` instead of `LOGLS_WARN`?  If it ever did occur in, say, qserv, it would indicate a programming error we'd want to notice and fix.

Reply:
Thank you for the explanation! I think it's a good idea to log this condition as `ERROR`.

Reply:
will do